core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -93,13 +93,26 @@ protected void killComputer(Computer c) {
     * Package-protected, but accessed API
     * ============================================================================================================== */
 
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    private Set<String> disabledAdministrativeMonitors = new HashSet<>();
 
-    @Restricted(NoExternalUse.class)
-    public CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
+    /**
+     * Get the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public Set<String> getDisabledAdministrativeMonitors(){
```

Comment:
https://github.com/search?q=org%3Ajenkinsci+getDisabledAdministrativeMonitors&type=Code

Reply:
So what's buggy about the `CopyOnWriteArraySet`?

Reply:
> https://github.com/search?q=org%3Ajenkinsci+getDisabledAdministrativeMonitors&type=Code

As expected, given it was `@Restricted`.

Reply:
> So what's buggy about the `CopyOnWriteArraySet`?

JCasC doesn't support it

Reply:
to avoid sharing the same data structure w/ the UI, why not just do something like this?
```
  synchronized(this.disabledAdministrativeMonitors) { return new HashSet<String>(this.disabledAdministrativeMonitors); };
```
... then any client of this class doesn't need to worry about concurrency concerns in the internal data structure; those concerns remain .. internal to this implementation

Reply:
because the returned set is modified by appending to it, it could be made immutable and then only modified via the setter instead possibly, if that would be better?

Reply:
could wrap w/ immutability. but what are you trying to guard against, really? is the intent that clients should fail fast when they try to make a modification that will never go anywhere? if so, then immutable is fine. otherwise, the i think the approach that i suggested already prevents the client from making accidental changes to an internal data structure

Reply:
It's to prevent concurrent modification exceptions. If you modify a non-threadsafe collection while iterating, it throws an exception.

Reply:
The issue is that clients modify this set, so returning a copy prevents them updating it

Reply:
ouch. any reason that clients can't be modified to invoke the setter instead of relying on directly mutating the internal state of this object? the current situation sounds like a pretty leaky design

Reply:
I suggested modifying it a couple of comments up, will take a look

Reply:
have changed, PTAL

Reply:
suggest wrapping this `return ...` statement w/ `synchronized(disabledAdministrativeMonitors) {}` as suggested above, since the `new HashSet<>(..)` operation will iterate over the collection passed to it